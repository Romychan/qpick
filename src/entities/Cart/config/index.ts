/** The name of the storage for storing data in the cart */
export const CART_STORAGE_NAME = 'cart';
