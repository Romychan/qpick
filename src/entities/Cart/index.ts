export { MOCK_CART, MOCK_CART_PRODUCT } from './lib/mock';

export { CartSummary } from './ui/CartSummary/CartSummary';

export { CartContext, CartContextProvider } from './model/context';
export { useCartStore } from './model/hooks';
