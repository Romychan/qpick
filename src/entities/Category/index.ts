export type { Category } from './model/types';
export { MAIN_CATEGORIES } from './model/constants';
