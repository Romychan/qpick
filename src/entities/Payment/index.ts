export { PAYMENT_TYPES } from './model/constants';
export {
  PaymentFormSchema,
  CardPaymentSchema,
  CashPaymentSchema,
} from './model/schemas';
export type {
  PaymentFormType,
  CardPaymentFormType,
  CashPaymentFormType,
} from './model/types';
