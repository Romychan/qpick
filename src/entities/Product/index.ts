export { ProductDetails } from './ui/ProductDetails/ProductDetails';
export { ProductRow } from './ui/ProductRow/ProductRow';
export { ProductCard } from './ui/ProductCard/ProductCard';

export type { Product } from './model/types';

export { MOCK_PRODUCTS } from './lib/mock';
