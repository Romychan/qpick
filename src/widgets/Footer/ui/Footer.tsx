import { Link } from 'react-router-dom';

import { Logo } from '~/shared/ui/Logo';
import { Card } from '~/shared/ui/Card';
import { SwitchLanguage } from '~/shared/ui/SwitchLanguage';
import { Menu } from '~/shared/ui/Menu';
import { MAIN_ROUTE } from '~/shared/config';

import {
  FIRST_FOOTER_MENU,
  FOOTER_LANGUAGES,
  SECOND_FOOTER_MENU,
} from '../model/constants';

import styles from './Footer.module.scss';
import { FooterSocial } from './FooterSocial/FooterSocial';

/**
 * This component is used to display the footer of a web page.
 * It contains a logo, navigation links, and a selection of languages.
 */
export const Footer = () => {
  return (
    <footer className={styles.container} data-testid="footer">
      <div className="container">
        <Card className={styles.inner}>
          <Link to={MAIN_ROUTE} className={styles.logo}>
            <Logo />
          </Link>

          <Menu direction="vertical" items={FIRST_FOOTER_MENU} />

          <div className={styles.wrapper}>
            <Menu direction="vertical" items={SECOND_FOOTER_MENU} />
            <SwitchLanguage languages={FOOTER_LANGUAGES} />
          </div>

          <FooterSocial />
        </Card>
      </div>
    </footer>
  );
};
