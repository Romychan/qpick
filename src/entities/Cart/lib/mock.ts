// eslint-disable-next-line @conarti/feature-sliced/layers-slices
import { Product } from '~/entities/Product/@x';

import { CartItem } from '../model/types';

/** Mock product for the cart */
export const MOCK_CART_PRODUCT: Product = {
  id: 1,
  title: 'Product',
  rate: 4.2,
  image: '/images/catalog/headphone-1.png',
  price: 1200,
  oldPrice: 1300,
  description: 'Test',
  category: 'Наушники',
};

/** List of mock cart items */
export const MOCK_CART: CartItem[] = [
  {
    product: MOCK_CART_PRODUCT,
    quantity: 2,
  },
];
