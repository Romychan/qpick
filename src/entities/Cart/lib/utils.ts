// eslint-disable-next-line @conarti/feature-sliced/layers-slices
import { Product } from '~/entities/Product/@x';

import { CartItem } from '../model/types';

/**
 * Add a product to the cart
 *
 * @param cartItems The current state of the cart
 * @param product Product to add to cart
 * @returns Updated state of the product cart
 */
export const addToCart = (cartItems: CartItem[], product: Product) => {
  const productInCart = cartItems.find(
    (item) => item.product.id === product.id,
  );

  if (!productInCart) {
    return [...cartItems, { product, quantity: 1 }];
  } else {
    return cartItems.map((item) =>
      productInCart.product.id === item.product.id
        ? { ...productInCart, quantity: item.quantity + 1 }
        : item,
    );
  }
};

/**
 * Delete a product from the cart
 *
 * @param cartItems The current state of the cart
 * @param id The ID of the product to be deleted from the cart
 * @returns Updated state of the product cart
 */
export const deleteFromCart = (cartItems: CartItem[], id: number) => {
  return cartItems.filter((item) => item.product.id !== id);
};

/**
 * Increase the number of product in the cart
 *
 * @param cartItems The current state of the cart
 * @param id Product ID to increase the quantity
 * @returns Updated state of the product cart
 */
export const increaseProductQuantity = (cartItems: CartItem[], id: number) => {
  const productInCart = cartItems.find((item) => item.product.id === id);

  if (productInCart) {
    return cartItems.map((item) =>
      productInCart.product.id === item.product.id
        ? { ...productInCart, quantity: item.quantity + 1 }
        : item,
    );
  }

  return cartItems;
};

/**
 * Decrease the number of product in the cart
 *
 * @param cartItems The current state of the cart
 * @param id Product ID to decrease the quantity
 * @returns Updated state of the product cart
 */
export const decreaseProductQuantity = (cartItems: CartItem[], id: number) => {
  const productInCart = cartItems.find((item) => item.product.id === id);

  if (productInCart) {
    return cartItems.map((item) =>
      productInCart.product.id === item.product.id
        ? { ...productInCart, quantity: item.quantity - 1 }
        : item,
    );
  }

  return cartItems;
};
