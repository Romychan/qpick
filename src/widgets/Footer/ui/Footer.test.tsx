import { RenderResult, screen } from '@testing-library/react';

import { CartContextProvider } from '~/entities/Cart';

import { renderWithRouter } from '~/shared/lib/tests';

import { Footer } from './Footer';

describe('Footer', () => {
  let container: RenderResult;

  beforeEach(() => {
    container = renderWithRouter(
      <CartContextProvider>
        <Footer />
      </CartContextProvider>,
    );
  });

  it('should render correctly', () => {
    const footerElement = screen.getByTestId('footer');

    expect(footerElement).toBeInTheDocument();
  });

  it('should match snapshot', () => {
    expect(container).toMatchSnapshot();
  });
});
