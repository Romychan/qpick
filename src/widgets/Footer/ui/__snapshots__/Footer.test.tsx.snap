// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Footer > should match snapshot 1`] = `
{
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <footer
        class="_container_794fa7"
        data-testid="footer"
      >
        <div
          class="container"
        >
          <div
            class="_container_1d0052 _inner_794fa7"
          >
            <a
              class="_logo_794fa7"
              href="/"
            >
              <div
                class="_container_4cc90f"
                data-testid="logo"
              >
                <p
                  class="_logo_4cc90f"
                >
                  Qpick
                </p>
              </div>
            </a>
            <nav
              class="_container_ca994d _vertical_ca994d"
            >
              <ul
                class="_list_ca994d"
              >
                <li
                  class="_container_cf770d"
                  data-testid="menu-item"
                >
                  <a
                    class="_link_cf770d"
                    href="/favorite"
                  >
                    Избранное
                  </a>
                </li>
                <li
                  class="_container_cf770d"
                  data-testid="menu-item"
                >
                  <a
                    class="_link_cf770d"
                    href="/cart"
                  >
                    Корзина
                  </a>
                </li>
                <li
                  class="_container_cf770d"
                  data-testid="menu-item"
                >
                  <a
                    class="_link_cf770d"
                    href="/contact"
                  >
                    Контакты
                  </a>
                </li>
              </ul>
            </nav>
            <div
              class="_wrapper_794fa7"
            >
              <nav
                class="_container_ca994d _vertical_ca994d"
              >
                <ul
                  class="_list_ca994d"
                >
                  <li
                    class="_container_cf770d"
                    data-testid="menu-item"
                  >
                    <a
                      class="_link_cf770d"
                      href="/terms"
                    >
                      Условия сервиса
                    </a>
                  </li>
                </ul>
              </nav>
              <div
                class="_container_b93fe0"
              >
                <svg
                  class="icon icon-planet _icon_b93fe0"
                  data-testid="icon"
                  height="20"
                  width="20"
                >
                  <use
                    xlink:href="/images/icons.svg#planet"
                  />
                </svg>
                <button
                  class="_button_0302f3 _button_b93fe0 _active_b93fe0 _link-primary_0302f3 _sm_0302f3"
                  data-testid="language-button"
                  type="button"
                >
                  Рус
                </button>
                <button
                  class="_button_0302f3 _button_b93fe0 _link-primary_0302f3 _sm_0302f3"
                  data-testid="language-button"
                  type="button"
                >
                  Eng
                </button>
              </div>
            </div>
            <nav
              class="_container_ca994d _horizontal_ca994d"
            >
              <ul
                class="_list_ca994d"
              >
                <li
                  class="_item_f235a0"
                >
                  <a
                    aria-label="VK"
                    class="_link_f235a0"
                    href="https://vk.com/"
                    rel="noopener noreferrer"
                    target="_blank"
                    title="VK"
                  >
                    <svg
                      class="icon icon-vk-filled filled"
                      data-testid="icon"
                      height="30"
                      width="30"
                    >
                      <use
                        xlink:href="/images/icons.svg#vk-filled"
                      />
                    </svg>
                  </a>
                </li>
                <li
                  class="_item_f235a0"
                >
                  <a
                    aria-label="Telegram"
                    class="_link_f235a0"
                    href="https://telegram.org/"
                    rel="noopener noreferrer"
                    target="_blank"
                    title="Telegram"
                  >
                    <svg
                      class="icon icon-telegram-filled filled"
                      data-testid="icon"
                      height="30"
                      width="30"
                    >
                      <use
                        xlink:href="/images/icons.svg#telegram-filled"
                      />
                    </svg>
                  </a>
                </li>
                <li
                  class="_item_f235a0"
                >
                  <a
                    aria-label="Whatsapp"
                    class="_link_f235a0"
                    href="https://www.whatsapp.com/"
                    rel="noopener noreferrer"
                    target="_blank"
                    title="Whatsapp"
                  >
                    <svg
                      class="icon icon-whatsapp-filled filled"
                      data-testid="icon"
                      height="30"
                      width="30"
                    >
                      <use
                        xlink:href="/images/icons.svg#whatsapp-filled"
                      />
                    </svg>
                  </a>
                </li>
              </ul>
            </nav>
          </div>
        </div>
      </footer>
    </div>
  </body>,
  "container": <div>
    <footer
      class="_container_794fa7"
      data-testid="footer"
    >
      <div
        class="container"
      >
        <div
          class="_container_1d0052 _inner_794fa7"
        >
          <a
            class="_logo_794fa7"
            href="/"
          >
            <div
              class="_container_4cc90f"
              data-testid="logo"
            >
              <p
                class="_logo_4cc90f"
              >
                Qpick
              </p>
            </div>
          </a>
          <nav
            class="_container_ca994d _vertical_ca994d"
          >
            <ul
              class="_list_ca994d"
            >
              <li
                class="_container_cf770d"
                data-testid="menu-item"
              >
                <a
                  class="_link_cf770d"
                  href="/favorite"
                >
                  Избранное
                </a>
              </li>
              <li
                class="_container_cf770d"
                data-testid="menu-item"
              >
                <a
                  class="_link_cf770d"
                  href="/cart"
                >
                  Корзина
                </a>
              </li>
              <li
                class="_container_cf770d"
                data-testid="menu-item"
              >
                <a
                  class="_link_cf770d"
                  href="/contact"
                >
                  Контакты
                </a>
              </li>
            </ul>
          </nav>
          <div
            class="_wrapper_794fa7"
          >
            <nav
              class="_container_ca994d _vertical_ca994d"
            >
              <ul
                class="_list_ca994d"
              >
                <li
                  class="_container_cf770d"
                  data-testid="menu-item"
                >
                  <a
                    class="_link_cf770d"
                    href="/terms"
                  >
                    Условия сервиса
                  </a>
                </li>
              </ul>
            </nav>
            <div
              class="_container_b93fe0"
            >
              <svg
                class="icon icon-planet _icon_b93fe0"
                data-testid="icon"
                height="20"
                width="20"
              >
                <use
                  xlink:href="/images/icons.svg#planet"
                />
              </svg>
              <button
                class="_button_0302f3 _button_b93fe0 _active_b93fe0 _link-primary_0302f3 _sm_0302f3"
                data-testid="language-button"
                type="button"
              >
                Рус
              </button>
              <button
                class="_button_0302f3 _button_b93fe0 _link-primary_0302f3 _sm_0302f3"
                data-testid="language-button"
                type="button"
              >
                Eng
              </button>
            </div>
          </div>
          <nav
            class="_container_ca994d _horizontal_ca994d"
          >
            <ul
              class="_list_ca994d"
            >
              <li
                class="_item_f235a0"
              >
                <a
                  aria-label="VK"
                  class="_link_f235a0"
                  href="https://vk.com/"
                  rel="noopener noreferrer"
                  target="_blank"
                  title="VK"
                >
                  <svg
                    class="icon icon-vk-filled filled"
                    data-testid="icon"
                    height="30"
                    width="30"
                  >
                    <use
                      xlink:href="/images/icons.svg#vk-filled"
                    />
                  </svg>
                </a>
              </li>
              <li
                class="_item_f235a0"
              >
                <a
                  aria-label="Telegram"
                  class="_link_f235a0"
                  href="https://telegram.org/"
                  rel="noopener noreferrer"
                  target="_blank"
                  title="Telegram"
                >
                  <svg
                    class="icon icon-telegram-filled filled"
                    data-testid="icon"
                    height="30"
                    width="30"
                  >
                    <use
                      xlink:href="/images/icons.svg#telegram-filled"
                    />
                  </svg>
                </a>
              </li>
              <li
                class="_item_f235a0"
              >
                <a
                  aria-label="Whatsapp"
                  class="_link_f235a0"
                  href="https://www.whatsapp.com/"
                  rel="noopener noreferrer"
                  target="_blank"
                  title="Whatsapp"
                >
                  <svg
                    class="icon icon-whatsapp-filled filled"
                    data-testid="icon"
                    height="30"
                    width="30"
                  >
                    <use
                      xlink:href="/images/icons.svg#whatsapp-filled"
                    />
                  </svg>
                </a>
              </li>
            </ul>
          </nav>
        </div>
      </div>
    </footer>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
